lightning/src/ln/msgs.rs

```diff
@@ -2448,13 +2457,22 @@ mod tests {
 	fn encoding_init() {
 		assert_eq!(msgs::Init {
 			features: InitFeatures::from_le_bytes(vec![0xFF, 0xFF, 0xFF]),
+			net_address: None,
 		}.encode(), hex::decode("00023fff0003ffffff").unwrap());
 		assert_eq!(msgs::Init {
 			features: InitFeatures::from_le_bytes(vec![0xFF]),
+			net_address: None,
 		}.encode(), hex::decode("0001ff0001ff").unwrap());
 		assert_eq!(msgs::Init {
 			features: InitFeatures::from_le_bytes(vec![]),
+			net_address: None,
 		}.encode(), hex::decode("00000000").unwrap());
+		assert_eq!(msgs::Init { features: InitFeatures::empty(),
+			net_address: Some(msgs::NetAddress::IPv4 {
+				addr: [127, 0, 0, 1],
+				port: 1000,
+			}),
+		}.encode(), hex::decode("0307017f000103e8").unwrap());
```

Comment:
The suggested hex here is just a suffix, you still need the same init message from the above tests as a prefix.